### DISCUSSION TITLE
Firebase auth on the server

### DISCUSSION THREAD

**Discussion by mitjans** [vuejs/vuefire#1329] [Questions and Help]
The Vuefire [documentation](https://vuefire.vuejs.org/nuxt/auth.html) states:

> Nuxt VueFire integrates with [Firebase Authentication](https://firebase.google.com/docs/auth) module to automatically synchronize the current user state on the server and the client.

I've tried to access `getCurrentUser()` on the server using Nuxt (`/server/api/endpoint.ts`), but the call to this function never returns and the server hangs.

I've been reading about SSR and firebase authentication on other frameworks, blog posts and videos. It seems that if you want to authenticate the user on the server, you need to [send an id token from the client and handle session cookies with Firebase admin SDK](https://www.youtube.com/watch?v=kRszxpeTnW0).

**Use case:** I want to access and modify a user's Firestore data from server side after a call to an API endpoint.

Is the handling of Auth server side the only way? Why is `getCurrentUser()` failing to return anything and hanging up the server entirely? Any plans on adding documentation on how to handle Auth in SSR and Nuxt?

Thanks!

**Comment by mitjans** [comment]
Supabase currently provides the mothod [serverSupabaseUser](https://supabase.nuxtjs.org/usage/services/server-supabase-user) that can be used in server endpoints. Is this planned for Vuefire as well?

**Comment by chiuwah** [comment]
I think you are mixing up two things. 1) The server side rendered (Nuxt) pages and 2) server routes (which uses the unjs/h3 http framework). VueFire is available in SSR part of Nuxt, but not available in server routes. You should use the Firebase Admin SDK for Firebase in server routes.

**Reply by posva** [reply]
This is true but we do have plans to add functions specific to server routes
